client/rest-high-level/src/main/java/org/elasticsearch/client/Request.java

```diff
@@ -139,8 +141,8 @@ static Request bulk(BulkRequest bulkRequest) throws IOException {
             bulkContentType = XContentType.JSON;
         }
 
-        byte separator = bulkContentType.xContent().streamSeparator();
-        ContentType requestContentType = ContentType.create(bulkContentType.mediaType());
```

Comment:
As a side note, I wonder how useful XContentType.mediaType() still is, except for JSON it redirects to mediaTypeWithoutParameters(). Maybe worth investigating if we can get rid of it in another issue.

Reply:
I agree - I'm currently looking at this.

Reply:
Would mix it with this PR though.